azure_costs_exporter/prometheus_collector.py

```diff
@@ -65,9 +66,15 @@ def _get_azure_data(self, month=None):
         url = "https://ea.azure.com/rest/{0}/usage-report?month={1}&type=detail&fmt=Json".format(self._enrollment,
                                                                                                  month)
 
-        rsp = requests.get(url, headers=headers, timeout=10)
-        rsp.raise_for_status()
+        try:
+            timeout = int(float(request.headers.get('Scrape-Timeout-Seconds', 10)))
+        except (RuntimeError, ValueError) as e:
```

Comment:
would it make sense to just write
```
except Exception:
```
It does not really matter what exception is raised. For any exception that might occur during the conversion, the default timeout should be used. Also, you don't use the `e`, so no point in naming the exception.